src/SmartComponents/DriftPage/FilterDropDown/FilterDropDown.js

```diff
@@ -9,44 +9,45 @@ class FilterDropDown extends Component {
     constructor(props) {
         super(props);
         this.onToggle = this.onToggle.bind(this);
-        this.getFilterbyState = this.getFilterByState.bind(this);
         this.createDropdownItem = this.createDropdownItem.bind(this);
     }
 
     onToggle() {
         this.props.toggleDropDown();
     }
 
-    getFilterByState(event, filter) {
-        this.props.changeFilter(filter);
-    }
-
-    createDropdownItem(display, value) {
-        let dropdownItem = <DropdownItem key={ value } component="button" onClick={ event =>
-            this.getFilterByState(event, value)
-        }>
-            { display }
-        </DropdownItem>;
+    createDropdownItem(stateFilter) {
+        let dropdownItem = <Checkbox label={ stateFilter.display } isChecked={ stateFilter.checked } onChange={ () =>
+            this.props.addStateFilter(stateFilter)
+        } />;
 
         return dropdownItem;
     }
 
     render() {
-        const dropdownItems = [
-            this.createDropdownItem('All', 'All'),
-            this.createDropdownItem('Same', 'SAME'),
-            this.createDropdownItem('Different', 'DIFFERENT'),
-            this.createDropdownItem('Incomplete Data', 'INCOMPLETE_DATA')
-        ];
+        const { stateFilters } = this.props;
+        const dropdownItems = [];
+        stateFilters.forEach(function(stateFilter) {
+            let dropdownItem = this.createDropdownItem(stateFilter);
+            dropdownItems.push(dropdownItem);
+        }.bind(this));
```

Comment:
Why does this have to bind?

Reply:
It doesn't know what this.createDropdownItem is if you don't. It says "this is undefined"

Reply:
I thought it should work if you drop the `this` all together.

Reply:
If I do that, it says "createDropdownItem" is undefined.